File: java/src/CFB.java
/**
 * @author egaeus
 * @mail sebegaeusprogram@gmail.com
 * @veredict Accepted
 * @url https://codeforces.com/problemset/problem/1353/B
 * @category implementation
 * @date 14/04/2020
 **/

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

import static java.lang.Integer.parseInt;

public class CFB {
    public static void main(String args[]) throws Throwable {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int T = parseInt(in.readLine());
        for (int t = 0; t < T; t++) {
            StringTokenizer st = new StringTokenizer(in.readLine());
            int N = parseInt(st.nextToken()), K = parseInt(st.nextToken());
            int[] A = new int[N];
            int[] B = new int[N];
            st = new StringTokenizer(in.readLine());
            for (int i = 0; i < N; i++)
                A[i] = parseInt(st.nextToken());
            st = new StringTokenizer(in.readLine());
            for (int i = 0; i < N; i++)
                B[i] = parseInt(st.nextToken());
            Arrays.sort(A);
            Arrays.sort(B);
            int result = 0;
            int i = N - 1, j = N - 1, I = 0;
            for (; I < N && K > 0; I++) {
                //System.out.println(i + " " + j + " " + K + " " + B[j] + " " + A[i]);
                if (j >= 0 && B[j] > A[i]) {
                    result += B[j];
                    j--;
                    K--;
                } else {
                    result += A[i];
                    i--;
                }
            }
            for (; I < N; I++, i--) result += A[i];
            System.out.println(result);
        }
    }
}

File: kotlin/src/CFB.kt
/**
 * @author egaeus
 * @mail sebegaeusprogram@gmail.com
 * @veredict Not sended
 * @url <https://codeforces.com/problemset/problem/>
 * @category ?
 * @date 29/05/2020
 **/

private fun readLn() = readLine()!! // string line
private fun readInt() = readLn().toInt() // single int
private fun readLong() = readLn().toLong() // single long
private fun readDouble() = readLn().toDouble() // single double
private fun readStrings() = readLn().split(" ") // list of strings
private fun readInts() = readStrings().map { it.toInt() } // list of ints
private fun readLongs() = readStrings().map { it.toLong() } // list of longs
private fun readDoubles() = readStrings().map { it.toDouble() } // list of doubles
private fun readString() = readLn() // single string

var S = CharArray(0)
var K1 = 0
var K2 = 0
var N = 0
var mem = arrayOfNulls<Array<Int>>(0)

fun main() {
    var T = readInt();
    repeat(T) {
        var ints = readInts()
        N = ints[0]
        K1 = ints[1]
        K2 = ints[2]
        S = readString().toCharArray()
        mem = arrayOfNulls(N)
        for (i in 0 until N)
            mem[i] = arrayOf(-1, -1)
        println(f(0, 0))
    }
}

fun f(p: Int, puseK: Int): Int {
    if (p >= N)
        return 0
    if (mem[p]!![puseK] >= 0)
        return mem[p]!![puseK]
    if (S[p] == '0')
        return f(p + 1, 0)
    var M = 0
    if (puseK == 1)
        M = f(p + 1, 0) + Math.min(K1, K2 - K1)
    else
        M = Math.max(f(p + 1, 0) + Math.min(K1, K2 - K1), f(p + 1, 1) + K1)
    mem[p]!![puseK] = M
    return M
}